Feat: adjust fields for ECS compatibility (#28)
logstash-plugins/logstash-input-exec: docs/index.asciidoc

Periodically run a shell command and capture the whole output as an event.

[NOTE]
========
* The `command` field of this event will be the command run.
* The `message` field of this event will be the entire stdout of the command.
========

> IMPORTANT: The exec input ultimately uses `fork` to spawn a child process.

Reviewer: Thank you for getting this extra changes in.  They render nicely and look good. Post-merge LGTM

----------------------------------
input {
  exec {
    command => "echo 'hi!'"
    interval => 30
  }
}
----------------------------------

This will execute `echo` command every 30 seconds.

[id="plugins-{type}s-{plugin}-ecs"]
==== Compatibility with the Elastic Common Schema (ECS)

This plugin adds metadata about the event's source, and can be configured to do so
in an {ecs-ref}[ECS-compatible] way with <<plugins-{type}s-{plugin}-ecs_compatibility>>.
This metadata is added after the event has been decoded by the appropriate codec,
and will not overwrite existing values.

|========
| ECS Disabled | ECS v1 , v8| Description

| `host`  | `[host][name]` | The name of the {ls} host that processed the event
| `command` | `[process][command_line]` | The command run by the plugin
| `[@metadata][exit_status]` | `[process][exit_code]` | The exit code of the process

| -- | `[@metadata][input][exec][process][elapsed_time]` | The elapsed time the command took to run in nanoseconds
| `[@metadata][duration]` | -- | Command duration in seconds as a floating point number (deprecated)

> |========

Reviewer 2: nit: plaintext table readability

```suggestion
|========
| ECS Disabled               | ECS v1 , v8               | Description

| `host`                     | `[host][name]`            | The name of the {ls} host that processed the event
| `command`                  | `[process][command_line]` | The command run by the plugin
| `[@metadata][exit_status]` | `[process][exit_code]`    | The exit code of the process
| --                         | `[@metadata][input][exec][process][elapsed_time]`
                                                         | The elapsed time the command took to run in nanoseconds
| `[@metadata][duration]`    | --                        | Command duration in seconds as a floating point number (deprecated)
|========
```

----------------------------------
input {
  exec {
    command => "echo 'hi!'"
    interval => 30
  }
}
----------------------------------

This will execute `echo` command every 30 seconds.

[id="plugins-{type}s-{plugin}-ecs"]
==== Compatibility with the Elastic Common Schema (ECS)

This plugin adds metadata about the event's source, and can be configured to do so
in an {ecs-ref}[ECS-compatible] way with <<plugins-{type}s-{plugin}-ecs_compatibility>>.
This metadata is added after the event has been decoded by the appropriate codec,
and will not overwrite existing values.

|========

> | ECS Disabled | ECS v1 | Description

Reviewer: ```suggestion
| ECS Disabled | ECS v1 , v8| Description
```

| `host`  | `[host][name]` | The name of the {ls} host that processed the event
| `command` | `[process][command_line]` | The command run by the plugin
| `[@metadata][exit_status]` | `[process][exit_code]` | The exit code of the process

> | `[@metadata][input][exec][process][elapsed_time]` | -- | The elapsed time the command took to run in nanoseconds

Reviewer 2: Implementation says this is [the opposite](https://github.com/logstash-plugins/logstash-input-exec/pull/28/files#diff-fdfe5b5dd3ccef8cc9228830f8800cc2bbfdd9f84ca59d15010357409dee7ed9R48)

```suggestion
| -- | `[@metadata][input][exec][process][elapsed_time]` | The elapsed time the command took to run in nanoseconds
```

Command to run. For example, `uptime`

[id="plugins-{type}s-{plugin}-ecs_compatibility"]
===== `ecs_compatibility`

  * Value type is <<string,string>>
  * Supported values are:
    ** `disabled`: uses backwards compatible field names e.g. `[host]`

>     ** `v1`, `v8`: uses fields that are compatible with ECS e.g. `[host][name]`

Reviewer: ```suggestion
    ** `disabled`: uses backwards compatible field names, such as `[host]`
    ** `v1`, `v8`: uses fields that are compatible with ECS, such as `[host][name]`
```

Controls this plugin's compatibility with the {ecs-ref}[Elastic Common Schema (ECS)].
See <<plugins-{type}s-{plugin}-ecs>> for detailed information.

> Example output:

Reviewer: ```suggestion
```
